### ISSUE TITLE
Mesh Shader Support

### ISSUE THREAD

**Issue by raytopianprojects** [panda3d/panda3d#1565]
## Description
Mesh shaders are a relatively new way of rendering 3d objects. They replace the default render pipeline and give developers much more control to how geometry is rendered. 

![image](https://github.com/panda3d/panda3d/assets/54505044/c818eac9-61ec-4e5b-a26a-79df4baa0689)

Effectively a compute shader for geometry they allow for things such as fine grained culling, procedural mesh generation, complex particle effects, and much more. Currently Vulkan, Metal, DirectX12, and OpenGL 4.6 (through a NVIDIA extension) all support mesh shaders. 

Besides just adding support for compiling and using mesh shaders Panda3D would need a way to generate and store meshlets/mesh clusters. https://github.com/zeux/meshoptimizer Is one way to generate meshlets before storing them in .bam.
 
https://on-demand.gputechconf.com/gtc-eu/2018/pdf/e8515-mesh-shaders-in-turing.pdf
https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/
https://www.khronos.org/blog/mesh-shading-for-vulkan
https://developer.nvidia.com/blog/using-mesh-shaders-for-professional-graphics/

## Use Case
My use case for mesh shaders would allow for both very high density scenes and the use of high polycount photogrammetry with less preprocessing (not needing to worry about baking assets ). 
![image](https://github.com/panda3d/panda3d/assets/54505044/67ba1bbd-8503-44c7-8c94-dac6dc256cc4)
![image](https://github.com/panda3d/panda3d/assets/54505044/9531103b-a41e-48c7-8085-edb15a47d1b8)
![image](https://github.com/panda3d/panda3d/assets/54505044/9f2e582f-b746-4d6e-ab8e-1ef710cf063d)

**Comment by Moguri** [comment]
Panda3D does not currently support Vulkan, Metal, nor D3D12, and all I can find for OpenGL is the Nvidia extension that looks like neither Intel nor AMD have implemented.

So, while I agree that mesh shader support will be nice in the future, it probably will not be added any time soon (at the very least we need to get the shader pipeline and Vulkan work wrapped up first). However, PRs are always welcome.

**Comment by raytopianprojects** [comment]
I agree this is certainly a feature for the post shader pipeline/vulkan future :smile:.  

As for bringing up the other graphics apis I am aware Panda3D doesn't support them. That's my bad for not being clear about why I brought them up which is that mesh shaders seem to be where the industry is heading long term and it'd be nice for Panda3D to support mesh shaders at some point. 

As for PRs I'll see if I can get something working with draw callback nodes first as a proof of concept. I'm not very good at C++ and I don't want to mess anything up😅.

**Comment by rdb** [comment]
For this to happen, a [cross-vendor OpenGL extension](https://github.com/KhronosGroup/OpenGL-Registry/pull/640) needs to be standardized.
